src/components/Button/Button.tsx

```diff
@@ -74,11 +87,34 @@ class Button extends UIComponent<IButtonProps, any> {
     classes,
     rest,
   }: IRenderResultConfig<IButtonProps>): ReactNode {
-    const { children, content } = this.props
+    const { children, content, icon, iconPosition, type } = this.props
+    const primary = type === 'primary'
+
+    const getContent = (): ReactNode => {
+      if (childrenExist(children)) {
+        return children
+      }
+
+      const iconIsAfterButton = iconPosition === 'after'
+      const renderedContent = [
+        content && <Text key="btn-content" truncated content={content} />,
+        icon &&
+          typeof icon === 'string' && (
+            <Icon
+              key="btn-icon"
+              name={icon}
+              xSpacing={!content ? 'none' : iconIsAfterButton ? 'before' : 'after'}
+              color={primary ? 'white' : 'black'}
+            />
+          ),
+      ].filter(Boolean)
+
+      return iconPosition === 'after' ? renderedContent : renderedContent.reverse()
+    }
```

Comment:
nit: could replace the condition with just `iconIsAfterButton`

Reply:
that was the point, I was sleeping probably; thx :)